### PR TITLE
Fix mAP problem in unit test of int8 object detection test

### PR DESCRIPTION
**Pull request by lidanqing-intel**
**mAP on full val set**
FP32 0.7363
INT8 0.7295

**Above mAP achieved on machine**
Intel(R) Xeon(R) Platinum 8180 CPU @ 2.50GHz. SkyLake

**Command to reproduce full val dataset (5000 images) :**
`/home/li/AIPG-Paddle/paddle/build/paddle/fluid/inference/tests/api/test_analyzer_int8_object_detection "ARGS" "--infer_model=/home/li/AIPG-Paddle/paddle/build/third_party/inference_demo/int8v2/mobilenet-ssd/model" "--infer_data=/home/li/.cache/paddle/dataset/pascalvoc/pascalvoc_full.bin" "--warmup_batch_size=10" "--batch_size=100" "--paddle_num_threads=4"`

related to issue #18913  Please review @wojtuss

### CODE REVIEW

**Comment by wojtuss** [comment]
@luotao1 , @qingqing01 ,
It turns out that comparing the accumulative mAP values requires 800 images for the test to pass. To avoid uploading a bigger dataset we could do the following:
1. Run only one iteration with batch size 100 and compare the current mAP values (which are outputs of index 1) - this way the test would still verify the accuracy properly on a small dataset.
2. Add an option (`--compare_acummulative_map=true`) so anyone who wants to run the test on a full dataset could switch to comparing accumulative mAP (the output of index 2). On the full dataset, the test passes with accumulative mAP comparison of course.

What do you think about it?

**Comment by lidanqing-vv** [comment]
Hi, @luotao1  @bingyanghuang 

**mAP on full val set**
FP32 0.7363
INT8 0.7295

**Above mAP achieved on machine**
Intel(R) Xeon(R) Platinum 8180 CPU @ 2.50GHz. SkyLake

**Command to reproduce full val dataset (5000 images) :**
`/home/li/AIPG-Paddle/paddle/build/paddle/fluid/inference/tests/api/test_analyzer_int8_object_detection "ARGS" "--infer_model=/home/li/AIPG-Paddle/paddle/build/third_party/inference_demo/int8v2/mobilenet-ssd/model" "--infer_data=/home/li/.cache/paddle/dataset/pascalvoc/pascalvoc_full.bin" "--warmup_batch_size=10" "--batch_size=100" "--paddle_num_threads=4"`

**Comment by lidanqing-vv** [comment]
PR_CI_CPU_Py2 fail 因为PADDLE_ENFORCE 没有被approve，但是我后来把PADDLE_ENFORCE 改成PADDLE_ENFORCE_LE之类的然后commit，这一个CI 还是挂。我就先revert到这一版了。 麻烦看一下谢谢！
这一版PR_CI_Coverage 通过了，因为我改paddle_enforce被重启了，但是它会过的

**Comment by luotao1** [comment]
> 但是我后来把PADDLE_ENFORCE 改成PADDLE_ENFORCE_LE之类的然后commit

修改文件中的所有`PADDLE_ENFORCE`都要做完修改。

**Comment by luotao1** [comment]
```

[00:19:24]	 42/164 Test #162: test_analyzer_int8_mobilenet_ssd ..............   Passed  134.27 sec
```
CI时间是可以的。 @bingyanghuang @wzzju 可以review

File: paddle/fluid/inference/tests/api/tester_helper.h [hunk 1]
```diff
@@ -443,15 +443,21 @@ void TestPrediction(const PaddlePredictor::Config *config,
   }
 }
 
-void SummarizeAccuracy(float avg_acc1_fp32, float avg_acc1_int8) {
+void SummarizeAccuracy(float avg_acc_fp32, float avg_acc_int8,
+                       int compared_idx) {
+  PADDLE_ENFORCE(compared_idx <= 2 && compared_idx >= 1,
+                 "Compare either top1 accuracy either mAP(top5), the "
```
Comment (wzzju) [inline paddle/fluid/inference/tests/api/tester_helper.h:449]: "Compare either top1 accuracy or mAP(top5), the "

File: paddle/fluid/inference/tests/api/analyzer_int8_object_detection_tester.cc [hunk 2]
```diff
@@ -273,7 +274,8 @@ TEST(Analyzer_int8_mobilenet_ssd, quantization) {
   q_cfg.mkldnn_quantizer_config()->SetWarmupData(warmup_data);
   q_cfg.mkldnn_quantizer_config()->SetWarmupBatchSize(FLAGS_warmup_batch_size);
 
-  CompareQuantizedAndAnalysis(&cfg, &q_cfg, input_slots_all);
+  // 0 is avg_cose, 1 is top1_acc, 2 is top5_acc or mAP
```
Comment (wzzju) [inline paddle/fluid/inference/tests/api/analyzer_int8_object_detection_tester.cc:277]: what is avg_cose?

**Comment by wzzju** [comment]
@bingyanghuang @luotao1 
The test results are moved here.

Command line:
`paddle/fluid/inference/tests/api/test_analyzer_int8_object_detection "ARGS" "--infer_model=third_party/inference_demo/int8v2/mobilenet-ssd/model" "--infer_data=/home/bingyang/.cache/paddle/dataset/pascalvoc/pascalvoc_full.bin" "--warmup_batch_size=10" "--batch_size=100" "--paddle_num_threads=1"`

- On Cascade Lake (New Machine) Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz:

```
I0814 20:39:33.458715 375748 tester_helper.h:468] --- Performance summary ---
I0814 20:39:33.458737 375748 tester_helper.h:469] FP32: avg fps: 52.8032, avg latency: 18.9383 ms
I0814 20:39:33.458747 375748 tester_helper.h:472] INT8: avg fps: 159.3869, avg latency: 6.2740 ms
I0814 20:39:33.458755 375748 tester_helper.h:452] --- Accuracy summary ---
I0814 20:39:33.458760 375748 tester_helper.h:453] Accepted mAP drop threshold: 0.01. (condition: (FP32_mAP  - INT8_mAP ) <= threshold)
I0814 20:39:33.458765 375748 tester_helper.h:457] FP32: avg mAP 0.7380
I0814 20:39:33.458768 375748 tester_helper.h:459] INT8: avg mAP 0.7291
```

- On Skylake (old machine) Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz:

```
I0814 17:46:16.489825  1646 tester_helper.h:468] --- Performance summary ---
I0814 17:46:16.489841  1646 tester_helper.h:469] FP32: avg fps: 45.7208, avg latency: 21.8719 ms
I0814 17:46:16.489859  1646 tester_helper.h:472] INT8: avg fps: 81.6199, avg latency: 12.2519 ms
I0814 17:46:16.489867  1646 tester_helper.h:452] --- Accuracy summary ---
I0814 17:46:16.489869  1646 tester_helper.h:453] Accepted mAP drop threshold: 0.01. (condition: (FP32_mAP  - INT8_mAP ) <= threshold)
I0814 17:46:16.489873  1646 tester_helper.h:457] FP32: avg mAP 0.7380
I0814 17:46:16.489876  1646 tester_helper.h:459] INT8: avg mAP 0.7303
```

File: paddle/fluid/inference/tests/api/tester_helper.h (hunk 1 above)
Comment (luotao1) [reply paddle/fluid/inference/tests/api/tester_helper.h:449]: PADDLE_ENFORCE_LE(compared_idx, 2, "xxx")
PADDLE_ENFORCE_GE(compared_idx,  1, "xxx")

**Pull request merged**